packages/wdio/src/command/test-id-index.command.ts

```diff
@@ -9,5 +10,6 @@ export const testID$$Index: ElsIndexSelectorFn = async (testID, index, context =
         throw new Error(`Cannot get item by testID "${testID}" with index "${index}"`);
     }
 
-    return elements[index];
+    // eslint-disable-next-line @typescript-eslint/return-await
+    return elements[index] as unknown as ChainablePromiseElement<WebdriverIO.Element>;
```

Comment:
Wouldn't it be better to change `testID$$` type to return `ChainablePromiseElement`